### PR TITLE
Modify course sets dialog

### PR DESCRIPTION
**Pull request by williamjallen**
Rearranged and re-styled the course sets dialog.  New green success button, more orderly list, and smaller header.

### CODE REVIEW

File: src/components/CourseSetEdit.vue [hunk 1]
```diff
@@ -12,70 +12,83 @@
...
-          class="open_close_icon, trash-btn"
-          @click="removeCourseSet(courseSet)"
-        ></font-awesome-icon>
-        {{ courseSet }}
-      </div>
+          <p
+            style="font-size: 80%;"
+            v-if="Object.keys(getCourseSets).length <= 1"
+          >
+            There must always be 1 course set, add another to remove the current
```
Comment (benjaminrsherman) [inline src/components/CourseSetEdit.vue:87]: Is this still needed?  The deletion icon only appears once there's at least two course sets.

File: src/components/CourseSetEdit.vue [hunk 2]
```diff
@@ -12,70 +12,83 @@
...
+                :disabled="!newCourseSetExists"
+                :class="{
+                  'btn-success': newCourseSetExists,
+                }"
+                :title="newCourseSetExists ? '' : 'Disabled'"
+                >Add Course Set</b-button
+              ></b-input-group-append
+            >
+            <b-form-invalid-feedback>
+              Must be a unique name
```
Comment (benjaminrsherman) [inline src/components/CourseSetEdit.vue:74]: Can we change this to be either `Must be a unique name` when the name conflicts or `Course sets can't have an empty name` (or something similar) when no input is given?

File: src/components/CourseSetEdit.vue [hunk 3]
```diff
@@ -12,70 +12,83 @@
...
-            "
-            :disabled="!newCourseSetExists"
-            :title="newCourseSetExists ? '' : 'Disabled'"
-            >Add Course Set</b-button
-          ></b-input-group-append
+      <ul class="list-group list-group-flush">
+        <li class="list-group-item">
+          <h5 class="mb-0">Course Sets:</h5>
+        </li>
+        <li
```
Comment (benjaminrsherman) [inline src/components/CourseSetEdit.vue:35]: These should be divs or have their styling changed so the table's row lines aren't visible and the spacing between consecutive course sets is significantly reduced.

File: src/components/CourseSetEdit.vue [hunk 4]
```diff
@@ -12,70 +12,83 @@
         @click="switchCurrentCourseSet(courseSet)"
         >{{ courseSet }}</b-dropdown-item
       >
+      <div class="dropdown-divider"></div>
       <b-dropdown-item v-b-modal.courseSet-modal>
         <font-awesome-icon
           title="Settings"
```
Comment (benjaminrsherman) [inline src/components/CourseSetEdit.vue:18]: The `title` for this icon should be `Edit Course Sets`.

File: src/components/CourseSetEdit.vue (hunk 3 above)
Comment (williamjallen) [reply src/components/CourseSetEdit.vue:35]: Why?  I deliberately switched to a table format with lines to give a little bit of separation between each element.  I could potentially inset the separator between each of the course sets slightly to make it clear that the course sets are one group.  I could also remove the left padding on the title and input field for a similar effect.

File: src/components/CourseSetEdit.vue (hunk 2 above)
Comment (williamjallen) [reply src/components/CourseSetEdit.vue:74]: I'll need to look into this some more since this depends on external libraries.

Comment (benjaminrsherman) [reply src/components/CourseSetEdit.vue:74]: You can make this reliant on a variable in the component, similar to how the Course Set name is rendered.

File: src/components/CourseSetEdit.vue (hunk 3 above)
Comment (benjaminrsherman) [reply src/components/CourseSetEdit.vue:35]: In my opinion they're too far apart.  It's fine if you want a bit of spacing between them, but it shouldn't be this pronounced.

**Comment by elihschiff** [comment]
can you try merging this with master. We fixed the submodules so that they no longer need you to commit `src/quacs-rs` or `src/store/data` so hopefully it will fix itself if you merge with master

**Pull request merged**